src/main/java/org/jenkins/plugins/lockableresources/LockableResourcesManager.java

```diff
@@ -501,9 +501,73 @@ public synchronized boolean reserve(List<LockableResource> resources,
 		return true;
 	}
 
+	private void unreserveResources(@Nonnull List<LockableResource> resources) {
+		for (LockableResource l : resources) {
+			l.unReserve();
+		}
+		save();
+	}
 	public synchronized void unreserve(List<LockableResource> resources) {
+		// make sure there is a list of resources to unreserve
+		if (resources == null || (resources.size() == 0)) {
+			return;
+		}
+		List<String> resourceNamesToUnreserve = new ArrayList<>();
 		for (LockableResource r : resources) {
-			r.unReserve();
+			resourceNamesToUnreserve.add(r.getName());
+		}
+
+		// check if there are resources which can be unlocked (and shall not be unlocked)
+		List<LockableResource> requiredResourceForNextContext = null;
+		QueuedContextStruct nextContext = this.getNextQueuedContext(resourceNamesToUnreserve, false);
```

Comment:
Okay, I'm honestly not sure: what if we're unreserving multiple resources and multiple job contexts can be unqueued and run?   This seems to be taken from unlockNames, so maybe there's something I'm missing that makes that okay -- or perhaps this is something that is missing too.

Would like to know the answer before we merge though (not sure if it's a bug or a subtlety here).